src/Stratis.Bitcoin.IntegrationTests.Common/SharedSteps.cs

```diff
@@ -69,7 +69,10 @@ public void MinePremineBlocks(CoreNode node, string walletName, string walletAcc
             this.WaitForNodeToSync(node);
 
             IEnumerable<UnspentOutputReference> spendable = node.FullNode.WalletManager().GetSpendableTransactionsInWallet(walletName);
-            Money amountShouldBe = node.FullNode.Network.Consensus.PremineReward + node.FullNode.Network.Consensus.ProofOfWorkReward;
+
+            // The premine and other block rewards are not mature yet.
+            Money amountShouldBe = 0;
```

Comment:
This isn't right as we need to ensure that the wallet has enough coins to stake with, otherwise the weight of the wallet will be too little to stake with.

Reply:
The premine is not immediately spendable. Clearly this method only mines sufficient blocks to generate the coins, nowhere near enough for them to be spendable, the comment indicates. I am correcting a constant that was chosen under false assumptions.

Reply:
Perhaps a better way would be to optionally include the immature coins and leave the calculation of the expected balance unchanged. It feels incorrect using a method that gets 'spendable' transactions to do that, though, so maybe a different or new method should be used instead.